test/src/test/java/hudson/model/JobTest.java

```diff
@@ -286,4 +291,35 @@ private static void tryConfigDotXml(JenkinsRule.WebClient wc, int status, String
         p.renameTo("different-name");
     }
 
+    @Issue("JENKINS-30502")
+    @Test public void testRenameTrimsSpace() throws Exception {
+        tryRename("myJob", " foo", "foo");
+        tryRename("myJob", "foo ", "foo");
+    }
+
+    private void tryRename(
+            String initialName,
+            String submittedName,
+            String correctResult) throws Exception {
+        j.jenkins.setCrumbIssuer(null);
+
+        FreeStyleProject job = createJobIfNeeded(initialName);
+        WebClient wc = j.createWebClient();
+        HtmlForm form = wc.getPage(job, "configure").getFormByName("config");
+        form.getInputByName("name").setValueAttribute(submittedName);
+        HtmlPage resultPage = j.submit(form);
+
+        String urlTemplate = "/job/{0}/rename?newName={1}";
+        String expectedURL = MessageFormat.format(
+                urlTemplate, initialName, correctResult);
+
+        assertTrue(resultPage.getUrl().toString().endsWith(expectedURL));
```

Comment:
add some human readable message either it will be difficult understand error message from test suite output.
Or even better use https://code.google.com/p/hamcrest/wiki/Tutorial `assertThat()`